src/workerd/jsg/jsg.c++

```diff
@@ -278,6 +278,29 @@ JsSymbol Lock::symbolAsyncDispose() {
   return IsolateBase::from(v8Isolate).getSymbolAsyncDispose();
 }
 
+void Lock::adjustExternalMemory(ssize_t amount) {
+  v8Isolate->AdjustAmountOfExternalAllocatedMemory(static_cast<int64_t>(amount));
+}
+
+namespace {
+struct ExternalMemoryAdjuster {
+  int64_t size;
+  v8::Isolate* isolate;
+  ExternalMemoryAdjuster(v8::Isolate* isolate, size_t size)
+      : size(static_cast<int64_t>(size)), isolate(isolate) {
+    isolate->AdjustAmountOfExternalAllocatedMemory(size);
+  }
+
+  ~ExternalMemoryAdjuster() noexcept(false) {
+    isolate->AdjustAmountOfExternalAllocatedMemory(-size);
+  }
```

Comment:
Review note: Obviously this can be a problem if the `kj::Own<void>` for this is held beyond the lifespan of the `isolate`. Should we include more protections against that by making the `isolate` effectively a weak ref rather than a bare pointer? Should we forgo this entirely in favor of just having the manual `adjustExternalMemory(...)` call?

The key use case for this would be to allow for something like..

```
auto ary = kj::heapArray<int>(10).attach(js.getExternalMemoryAdjuster(10 * sizeof(int)));
```

Such that the external memory will be automatically adjusted in the isolate when the array is freed. This, of course, is only really appropriate if the array does not get bound to an `ArrayBuffer` at any point, which would cause the data to be double accounted.